### ISSUE TITLE
java.lang.IllegalStateException: You need to add at least one AlgoliaResultsListener.

### ISSUE THREAD

**Issue by alisonthemonster** [algolia/instantsearch-android#145]
**Describe the bug 🐛**
Hi there! I'm trying to set up an InstantSearch widget for a project following the [Getting Started](https://www.algolia.com/doc/guides/building-search-ui/getting-started/android/) guide. The only difference in the project is that I'm using a Fragment instead of an Activity for my search view. 
 
Here is my code:
Fragment OnCreateView:
```
        searcher = Searcher.create("latency", "3d9875e51fbd20c7754e65422f7ce5e1", "bestbuy")
        val helper = InstantSearch(activity!!, searcher)
        helper.search()
```

And in my OnDestroy
```
        searcher.destroy()
```

My layout has `com.algolia.instantsearch.ui.views.Hits`  and I have a layout for each of the hit items. That layout has the `layout` tag around it for databinding and databinding is enabled (although its through the new way of enabling databinding): 
```
android {
    ...
    dataBinding {
        enabled = true
    }
}
```

From the instructions it seems like I should be all set from there but when I launch the app I get this error:
```
java.lang.IllegalStateException: You need to add at least one AlgoliaResultsListener.
        at com.algolia.instantsearch.ui.helpers.InstantSearch.processAllListeners(InstantSearch.java:401)
        at com.algolia.instantsearch.ui.helpers.InstantSearch.processActivity(InstantSearch.java:353)
        at com.algolia.instantsearch.ui.helpers.InstantSearch.<init>(InstantSearch.java:89)
        at <MyFragment>.onCreateView(MyFragment.kt:57)
        at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2440)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:885)
        at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:2078)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1852)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1808)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1709)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2609)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2565)
        at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:2467)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:908)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1229)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1295)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2605)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2565)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:245)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:525)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:179)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
        at android.app.Activity.performStart(Activity.java:7165)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

I looked through your examples and only saw one instance of `AlgoliaResultsListener`  on your MapWidget in the MapExample. Am I missing a piece of code somewhere? Or do Fragments need to set up InstantSeach differently?

Thank you!

**Environment:**
 - OS: android
 - Version:  `com.algolia:instantsearch-androidx:1.15.2`

**Comment by q-litzler** [comment]
Hello,

This exceptions means that your `InstantSearch` instance could not find any listeners attached to it. An `AlgoliaResultsListener` is usually implemented by a `Widget`.
Can you please use the `registerWidget` method on your `Hits` view, and tell us if you are still facing the same problem ?

Thank you.

**Comment by alisonthemonster** [comment]
Thanks for the reply! I think I figured it out, since I'm using fragments I think it was having problems finding my Hits because it was looking for them in the Activity. When I switched over to this constructor: `InstantSearch(search_results, searcher)` where `search_results` is my Hits view I no longer got the error!

Now my code looks like:
```
searcher = Searcher.create(ALGOLIA_APP_ID, ALGOLIA_SEARCH_API_KEY, ALGOLIA_INDEX_NAME)
val helper = InstantSearch(search_results, searcher)
helper.search()
```
I also made sure to have this code in my `onViewCreated` and now it works!
